Automattic/jetpack
# E2E Tests: Add more reporting

This PR is a bit of mess since it fixes/stabilizes connection flow, and also uses better slack reporter

#### Changes proposed in this Pull Request:

- Don't wait for a progress bar at all in thank you page
- disable `no-console` eslint rule
- introduce test failure interceptor to be able to take screenshots and send slack messages
- add the ability to log page HTML on test failures for debugging purposes 
- Update screenshot reporter
- Use Slack web API for sending formatted messages and upload images into the slack channel
- re-generate encrypted config file with slack related secrets: `E2E_SLACK_TOKEN`, `E2E_CHANNEL_NAME`

#### Is this a new feature or does it add/remove features to an existing part of Jetpack?
* If you're an Automattician, include a shortlink to the p2 discussion with Jetpack Product here.
p9dueE-Ju-p2

#### Testing instructions:






* Check the Travis build log
* Try to run it locally: `WP_BASE_URL="YOUR_NOT_CONNECTED_JETPACK_SITE" WP_PASSWORD="wordpress" WP_USERNAME="wordpress" PUPPETEER_HEADLESS=false yarn test-e2e`

Here is an example of slack message: p1560429675025200-slack-C1Y1F7ZM1

#### Proposed changelog entry for your changes:

* N/a
